Managers/ConsoleMan.cpp

```diff
@@ -1,541 +1,423 @@
+		PrintString("CTRL + P - Show performance stats");
+		PrintString("CTRL + R - Reset activity");
+		PrintString("CTRL + M - Switch display mode: Draw -> Material -> MO");
+		PrintString("CTRL + O - Toggle one sim update per frame");
+		PrintString("----------------");
+		PrintString("F2 - Reload all Lua scripts");
+		PrintString("F3 - Save console log");
+		PrintString("F4 - Save console user input log");
+		PrintString("F5 - Clear console log ");
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ConsoleMan::Update() {
+		if (g_UInputMan.FlagCtrlState() && g_UInputMan.KeyPressed(KEY_TILDE)) {
+			SetReadOnly();		
+		}
+
+		if (!g_UInputMan.FlagShiftState() && (!g_UInputMan.FlagCtrlState() && g_UInputMan.KeyPressed(KEY_TILDE))) {
+			if (IsEnabled()) {
+				if (!m_ReadOnly) {
+					m_InputTextBox->SetEnabled(false);
+					m_GUIController->GetManager()->SetFocus(0);
+					// Save any text being worked on in the input, as the box keeps getting junk added to it
+					m_LastInputString = m_InputTextBox->GetText();
+					SetEnabled(false);
+				} else {
+					// Restore any text being worked on in the input and set the focus and cursor position of the input line
+					m_InputTextBox->SetText(m_LastInputString);
+					m_InputTextBox->SetCursorPos(m_InputTextBox->GetText().length());
+					m_ReadOnly = false;
+				}
+			} else {
+				m_InputTextBox->SetText(m_LastInputString);
+				m_InputTextBox->SetCursorPos(m_InputTextBox->GetText().length());
+				SetEnabled(true);
+			}
+		}
+
+		if (m_ConsoleState != ENABLED && m_ConsoleState != DISABLED) { ConsoleOpenClose(); }
+
+		if (m_ConsoleState != ENABLED) {
+			return;
+		}
+
+		m_GUIController->Update();
+
+		// Resize console using Ctrl+Up/Down
+		if (g_UInputMan.FlagCtrlState() && g_UInputMan.KeyPressed(KEY_DOWN)) {
+			SetConsoleScreenSize(m_ConsoleScreenRatio + 0.05F);
+		} else if (g_UInputMan.FlagCtrlState() && g_UInputMan.KeyPressed(KEY_UP)) {
+			SetConsoleScreenSize(m_ConsoleScreenRatio - 0.05F);
+		}
+
+		if (!m_ReadOnly) {
+			m_InputTextBox->SetEnabled(true);
+			m_InputTextBox->SetFocus();
+	
+			// Get rid of any grave accents (`) that are pasted/typed into the textbox.
+			std::string textBoxString = m_InputTextBox->GetText();
+			if (std::find(textBoxString.begin(), textBoxString.end(), '`') != textBoxString.end()) {
+				textBoxString.erase(std::remove(textBoxString.begin(), textBoxString.end(), '`'), textBoxString.end());
+				m_InputTextBox->SetText(textBoxString);
+				m_InputTextBox->SetCursorPos(textBoxString.length());
+			}	
+		} else {
+			m_InputTextBox->SetEnabled(false);
+			m_GUIController->GetManager()->SetFocus(0);
+			return;
+		}	
+
+		if (g_UInputMan.KeyPressed(KEY_ENTER) || g_UInputMan.KeyPressed(KEY_ENTER_PAD)) { FeedString(); }
+
+		// If multiple strings were pasted into the box, we need to check for and submit all the strings
```

Comment:
I don't think we need this comment, the method you pulled out explains it just fine.